plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -3279,7 +3282,14 @@ public Optional<ConstraintApplicationResult<ConnectorTableHandle>> applyFilter(C
             for (Entry<ColumnHandle, Domain> domainEntry : constraintDomains.entrySet()) {
                 DeltaLakeColumnHandle column = (DeltaLakeColumnHandle) domainEntry.getKey();
                 if (!partitionColumns.contains(column)) {
-                    unenforceableDomains.put(column, domainEntry.getValue());
+                    if (column.equals(fileModifiedTimeColumnHandle())) {
```

Comment:
The column: `$path` and `$file_size` also can be supported

Reply:
Are you handling these use cases in a follow up?

Reply:
Yes, just added.

Reply:
Add as well pls

<code>
Use a `WHERE` clause with [metadata columns](delta-lake-special-columns) to filter
which files are optimized.

```sql
ALTER TABLE test_table EXECUTE optimize
WHERE "$file_modified_time" > date_trunc('day', CURRENT_TIMESTAMP);
```
</code>

to `delta-lake.md`.